test/Microsoft.Framework.ConfigurationModel.Json.Test/ArrayTest.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.Framework.ConfigurationModel.Test;
+using Xunit;
+
+namespace Microsoft.Framework.ConfigurationModel.Json.Test
+{
+    public class ArrayTest
+    {
+        [Fact]
+        public void ArraysAreConvertedToKeyValuePairs()
+        {
+            var json = @"{
+                'ip': [
+                    '1.2.3.4',
+                    '7.8.9.10',
+                    '11.12.13.14'
+                ]
+            }";
+
+            var jsonConfigSource = new JsonConfigurationSource(TestStreamHelpers.ArbitraryFilePath);
+            jsonConfigSource.Load(TestStreamHelpers.StringToStream(json));
+
+            Assert.Equal("1.2.3.4", jsonConfigSource.Get("ip:0"));
+            Assert.Equal("7.8.9.10", jsonConfigSource.Get("ip:1"));
+            Assert.Equal("11.12.13.14", jsonConfigSource.Get("ip:2"));
```

Comment:
the configuration source doesn't expose the number of keys. But I will add boundary checks for those tests that use `IConfiguration`